of_design_doc.md

```diff
@@ -39,33 +41,59 @@ The system encodes extra useful information into type parameters:
 - `flatten(T::Type{<:OfType}, values)` - Convert structured values to flat vector
 - `unflatten(T::Type{<:OfType}, vec)` - Reconstruct structured values from flat vector
 
-### 4. Symbolic Dimensions with Constants
+### 4. The @of Macro
+
+The `@of` macro provides cleaner syntax by automatically converting field references to symbols:
+
+```julia
+# Instead of writing:
+T = of((
+    n = of(Int; constant=true),
+    data = of(Array, :n, 2)
+))
+
+# You can write:
+T = @of(
+    n = of(Int; constant=true),
+    data = of(Array, n, 2)  # 'n' is automatically converted to :n
+)
+```
+
+### 5. Symbolic Dimensions and Bounds
 
 For cases where dimensions need to be specified at runtime:
 
 ```julia
 # Define type with symbolic dimensions
 MatrixType = of((
```

Comment:
I suggest we remove support for using `of((;...))` and only keep `@of()` for composable types for simplicity and to reduce the maintenance burden.